File: grind/Easy/E104.java
package grind.Easy;

import grind.TreeNode;

// max depth of binary tree
// runtime: 100%, memory: 93%


public class E104 {
    public int maxDepth(TreeNode root) {
        // recursivly perform DFS to determine max depth
        if(root == null){
            return 0;
        }else{
            return Math.max(maxDepth(root.left), maxDepth(root.right)) + 1;
        }
    }
}

File: grind/Easy/E104.go
package easy

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

/*
Get the max depth of a binary search tree

runtime: 67%, memory: 75%
*/

func maxDepth(root *TreeNode) int {
	return helper(root)
}

func helper(root *TreeNode) int {
	if root == nil {
		return 0
	} else {
		return max(helper(root.Left), helper(root.Right)) + 1
	}
}
